lib/vtls/schannel.c

```diff
@@ -558,7 +577,7 @@ schannel_connect_step2(struct connectdata *conn, int sockindex)
     memcpy(inbuf[0].pvBuffer, connssl->encdata_buffer,
            connssl->encdata_offset);
 
-    host_name = Curl_convert_UTF8_to_tchar(hostname);
+    host_name = Curl_convert_UTF8_to_tchar(conn->host.name);
```

Comment:
Same as above, this might break HTTPS-proxy support. Please elaborate on the reason for this change.